### PR TITLE
Add error check for 'messagesopt'

### PR DESCRIPTION
**Pull request by h-east**
Add a check to "wait:" value is less than 0.
Similar to "history:".

### CODE REVIEW

**Comment by zeertzjq** [comment]
This can also be solved by using `long`:
```diff
diff --git a/src/message.c b/src/message.c
index f0d1806c6..f99ad2b16 100644
--- a/src/message.c
+++ b/src/message.c
@@ -1090,8 +1090,8 @@ messagesopt_changed(void)
 {
     char_u	*p;
     int		messages_flags_new = 0;
-    int		messages_wait_new = 0;
-    int		messages_history_new = 0;
+    long	messages_wait_new = 0;
+    long	messages_history_new = 0;
 
     p = p_mopt;
     while (*p != NUL)
@@ -1141,8 +1141,8 @@ messagesopt_changed(void)
     if (messages_wait_new > 10000)
         return FAIL;
 
-    msg_flags = messages_flags_new;
-    msg_wait = messages_wait_new;
+    msg_flags = (int)messages_flags_new;
+    msg_wait = (int)messages_wait_new;
 
     msg_hist_max = messages_history_new;
     check_msg_hist();
```
as `getdigits()` uses `atol()`, which returns `LONG_MAX` on overflow.

**Comment by Shougo** [comment]
I have used `int` for the values.  Because `diffopt_changed()` uses `int`...

**Comment by h-east** [comment]
There is no problem with "(signed) int". In fact, signed is more convenient for checking overflow and underflow.

**Comment by h-east** [comment]
@zeertzjq 
> which returns LONG_MAX on overflow.

Really?

From [atoi(3) — Linux manual page](https://man7.org/linux/man-pages/man3/atoi.3.html)
```
NAME
       atoi, atol, atoll - convert a string to an integer
(snip...)
BUGS
       (snip...)
       No checks for overflow or underflow are done.
```

**Comment by zeertzjq** [comment]
> @zeertzjq
> 
> > which returns LONG_MAX on overflow.
> 
> Really?
> 
> From [atoi(3) — Linux manual page](https://man7.org/linux/man-pages/man3/atoi.3.html)
> 
> ```
> NAME
>        atoi, atol, atoll - convert a string to an integer
> (snip...)
> BUGS
>        (snip...)
>        No checks for overflow or underflow are done.
> ```

Oops, I skipped that part and took the `LONG_MAX` from `strtol()`.

**Comment by chrisbra** [comment]
Thanks. I also add a test for negative values.
